src/utils.ts

```diff
@@ -33,14 +38,23 @@ interface PropertyFilteringAction {
   type: 'property-filtering';
   query: PropertyFilterQuery;
 }
-type Action<T> = SelectionAction<T> | SortingAction<T> | PaginationAction | FilteringAction | PropertyFilteringAction;
+type Action<T> =
+  | SelectionAction<T>
+  | ExpandAction<T>
+  | SortingAction<T>
+  | PaginationAction
+  | FilteringAction
+  | PropertyFilteringAction;
 export type CollectionReducer<T> = Reducer<CollectionState<T>, Action<T>>;
 export function collectionReducer<T>(state: CollectionState<T>, action: Action<T>): CollectionState<T> {
   const newState = { ...state };
   switch (action.type) {
     case 'selection':
       newState.selectedItems = action.selectedItems;
       break;
+    case 'expand':
```

Comment:
If I understand this correctly, this action type is used for both expand and collapse, and also, the other action types in this reducer use nouns instead of verbs. Should this better be called something like `expansion`, `expansionState`, `expandedState` or `expandCollapseState`?

Reply:
Makes sense, will rename!